Node Status Monitor/NodeJS/node_monitor.js

```diff
@@ -0,0 +1,197 @@
+/* imports*/
+import fetch from 'node-fetch';
+import fs from 'fs';
+import log from './utils/logger.js';
+import { get_token } from './api/auth.js';
+import config from './config/config.js';
+
+
+/* 
+variables:
+        Key                             Description
+    Only_changed_data :  if true get only changed node status, all node with status
+    
+functions:
+        Function Name                   Description
+    fetch_Data            :     function to fetch Node Details
+    process_node_status   :     function to process Node Details
+    write_data            :     function to write Node Details
+    read_old_data         :     function to read old Node Details
+    compare_data          :     function to compare  Node Details 
+    get_token             :     function to get Access token 
+    
+Files:
+        FileName                        Description
+    new_node_data.json    :      Json Format Result with Node's changed Status
+    old_node_data.json    :      Json Format Data Node's status Previous run
+    result.json         :      Json Format Data Node's status Current run  
+*/
+
+
+
+
+/* Gobal Variable */
+var only_changed_data = true; // if true get only changed node status, all node with status
+
+let nodes_detail_url = config.nodes_detail_url;
+let client_key = config.client_key;
+let client_secret = config.client_secret;
+// files
+let new_data_file = config.files.new_data_file;
+let old_data_file = config.files.old_data_file;
+let result_file = config.files.result_file;
+
+/* function call */
+get_node_details();
+
+
+async function get_node_details() {
+    try {
+        let token = await get_token(client_key, client_secret);
+        let get_new_node_details = await fetch_Data(token);
+        let new_node_details = await process_node_status(get_new_node_details);
+        let old_node_details = await read_old_data();
+        if (!old_node_details) {
+            write_data(new_node_details);
+        } else {
+            compare_data(old_node_details, new_node_details);
+        }
+        return new_node_details
+    }
+    catch (err) {
+        let error = new Error(err);
+        log.error(error);
+    }
+}
+
+/*function to fetch Node Details */
+function fetch_Data(token) {
+    return new Promise((resolve, reject) => {
+        log.info("-------------------- fetching Node Details --------------------")
+        fetch(nodes_detail_url, {
+            headers: {
+                'accept': 'application/json',
+                'content-length': '0',
+                'authorization': `Bearer ${token}`
+            }
+        })
+            .then(res => res.json())
+            .then(json => {
+                // if object has propeery Message ,display Error, else Process Data
+                if (json.hasOwnProperty('Message')) {
+                    log.error(`************************* ${json.Message} *************************`);
```

Comment:
In case of multi line message, do not wrap in `*` characters.
```suggestion
                    log.error(json.Message);
```

Reply:
Done!